### ISSUE TITLE
Problems about the code 

### ISSUE THREAD

**Issue by SMZCC** [bilylee/SiamFC-TensorFlow#18]
#### Dear @bilylee
  Hello
##### Thanks for your contributions :tada: 
I have encountered a problem in the code which lies in [line](https://github.com/bilylee/SiamFC-TensorFlow/blob/17f51563f081281fc7d966602d393715e7ecb469/metrics/track_metrics.py#L94) .
- my problem is that how does this function work ?
- to understand it , I have written a code snippet(mimic your function) as the following:
```python
    def get_mean(values):
        with tf.name_scope('mean'):
            mean, update_op = tf.metrics.mean(values)  # values = [1, 2, 3, 4, 5]
            with tf.control_dependencies([update_op]):
                mean = tf.identity(mean)
            return mean

    g = tf.Graph()
    g.as_default()
    values = tf.convert_to_tensor([1, 2, 3, 4, 5], dtype=tf.float32)
    mean= get_mean(values)
    init_global_variables = tf.global_variables_initializer()
    init_local_variables = tf.local_variables_initializer()
    g.finalize()

    sess = tf.InteractiveSession(graph=tf.get_default_graph())
    sess.run(init_global_variables)
    sess.run(init_local_variables)
    print sess.run(values)
    for i in range(5):
        print 'mean:', sess.run(mean)

# out:
# [1. 2. 3. 4. 5.]
# mean: 0.0
# mean: 6.0
# mean: 4.5
# mean: 4.0
# mean: 3.75

# not corresponding to my expectation
# expected values are:
# [1. 2. 3. 4. 5.]
# mean: 3.0
# mean: 3.0
# mean: 3.0
# mean: 3.0
# mean: 3.0

# reasons as following：
# iter: 1
# total = 1+2+3+4+5 = 15
# count = 1+1+1+1+1 = 5
# mean = 15 / 5 = 3
# iter: 2
# total = 15 + 15 = 30
# count = 5 + 5 = 10
# mean = 30 / 10 = 3
...
```
- Would you please told me I did anything wrong ? And how to understand it ?
#### Best regards

**Comment by PeterHuang2015** [comment]
Hi, @SMZCC  Here's what I got, different from yours,
![2018-04-23 9 59 30](https://user-images.githubusercontent.com/13297503/39103401-19082510-46dd-11e8-8010-01602911616e.png)
as a personal opinion, the function aims to calculate how many response maps has a peak outside the manually labels center which makes it simple and intuitive, your question may lay on mechanisms of tensorflow. As for this issue, I suggest reading the official document and report your experiment settings here(os, tf version, python version, etc.), while I also got 0.0 in the first round (tf=1.4,cpu, py=2.7), I will be better to double check the step in execution.

**Comment by bilylee** [comment]
This parameter is the stride of the feature extractor. It is used for computing ground truth labels during training and target positions during testing.

I used `convolutional_alexnet.stride = 8` in the early days of implementing SiamFC. Its usage is now replaced by https://github.com/bilylee/SiamFC-TensorFlow/blob/17f51563f081281fc7d966602d393715e7ecb469/configuration.py#L36

You can safely delete this line without affecting anything.

**Comment by SMZCC** [comment]
> You can safely delete this line without affecting anything.

I have tried and I found nothing changed , which made me confused, but now , it's OK.Thanks a lot !
#### Good job :+1: :muscle: @bilylee 
#### Best regards
